### DISCUSSION TITLE
Copilot not working with MacOS Spoken Content as expected

### DISCUSSION THREAD

**Discussion by shawnthompson** [community/community#120119] [Accessibility]
### Select Topic Area

Bug

### Body

I'd like to report an issue I'm having with **Github** Copilot and my [MacOS Spoken Content](https://support.apple.com/en-ca/guide/mac-help/mh27448/mac) accessibility feature I use.

When I try to get Spoken Content to read the **Github** Copilot chat panel, it won't read the text as expected. I have to select the text and get it to read it by the system. This is not the behaviour I expect it to have.

I created this short little video to demonstrate the issue. Enjoy 🍿 

[GitHub Copilot with MacOS Spoken Content - YouTube](https://www.youtube.com/watch?v=WUX4Czg1TMo&feature=youtu.be)

I had the same problem with Slack messages and they were able to fix it after I reported it to them. I believe it has something to do with Electron. The same problem with MS Teams when it was updated also, I reported it but not sure if that's going anywhere 🤔.

**Comment by queenofcorgis** [comment]
Thank you for this report and the video 😄 . It has been logged internally.

Any additional members who are experiencing the same situation, please comment to share or upvote it.

We appreciate you contributing and continuing to support other users in the GitHub Community Discussions.

**Comment by cehfisher** [comment]
Hi @shawnthompson - I asked one of GitHub's internal accessibility team members to review your request and received the following response.

> I do not believe this is a WCAG violation nor a usability issue. I'd say it's a user-agent/compatibility issue.

> I confirmed that MacOS Spoken Content works as demonstrated with Safari as in the YouTube video (clicking text then Opt+Esc to activate the reading aloud). BUT, Spoken Content doesn't work this way with the Chrome browser. One must select the text, not click.

> GitHub Copilot in VS Code is developed with Electron which uses the Chromium/Blink browser engine. That explains why the user can't click text to use the spoken content feature (since Chrome also runs on the Chromium/Blink browser engine).

> Furthermore, options for MacOS Spoken Content don't include clicking text. Only selecting text. So I don't feel there's a bug here. The Safari behavior seems to be a bonus for the user. At the most it could possibly be is a usability improvement in support between Chromium/Blink and macOS Spoken Content.

Does this answer help explain the incompatibility? Or do you need additional support at this time?

As a next step, you may consider reaching out to the Chrome or VS Code teams to submit a feature request to include this type of spoken content support.

Thanks,
Carie

**Reply by shawnthompson** [reply]
Passing the buck I see 👅 . I guess I'll go on the hunt to report it to Chrome or VS Code. I'll be referencing this discussion. 

<span lang="fr">Merci beaucoup</span>,
Shawn

**Reply by cehfisher** [reply]
I do wish we had the ability to make the changes you requested, as it certainly seems like a feature others would want as well. Let me know if you need any a11y support, should you go the Chrome/VS Code route. Thank you for reaching out Shawn! 😄

**Comment by shawnthompson** [comment]
This is the answer I got from Slack when I asked them how they fixes it.

> We did notice, with your report and a few others, that the behavior had regressed, and determined that Spoken Content worked when we were based on Electron version 26 and then regressed in version 27. It was fixed in version 29.0.0 which will rolled out with Slack version 4.37.
